src/utils/desktop.js

```diff
@@ -147,9 +148,13 @@ export const resourceResolver = (core) => {
 
   const soundsEnabled = () => !!getSoundThemeName();
 
-  const icon = path => {
+  const icon = (name) => {
```

Comment:
Maybe extensions should be removed here ? I imagine they are supplied in other places, so doing removal will ensure backward compatability.

You can probably add tests for this as well.

Reply:
Do you just want me to remove everything after/including the last period if it exists?

Reply:
If the extension is not removed then this function would fail if you gave it `foo.png` as it would resolve to `foo.png.png`.

Reply:
You can see this if you add some tests :)

Reply:
I am aware of the double extension flaw; I was asking how you want that handled because of the edge case we discussed in `dev-meta` with the dotted file name issue. Removing everything after the last period could be problematic with an image name such as `windows.start.icon` for individuals who name things according to that format, so I'm unsure of the approach you would like to take to mitigate the issue.

I don't see the passing of an extension as much of an issue, I only noticed one place that used it and that was fixed in this PR (was for the taskbar search icon). So anyone who is updating will have the fix anyways. Regardless, I'm good with adding in a fix for extension compatibility if you have a suggestion of how to implement it.

And yes I will be adding tests 👍🏼

Reply:
You can just use a list of known extensions and remove that. Chances are that someone naming stuff `icon.png.png` is slim :)

Reply:
> Chances are that someone naming stuff

If this is the case then just slapping on some notices in the documentation would be the way to go. Maybe enforce some kind of pattern (which is why I went for GTK icon packs) ?